src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java

```diff
@@ -0,0 +1,160 @@
+  }
+
+  @Before(order = 1, value = "@GCS_TEXT_TEST")
+  public static void createBucketWithTextFile() throws IOException, URISyntaxException {
+    gcsSourceBucketName = createGCSBucketWithFile(PluginPropertyUtils.pluginProp("gcsTextFile"));
+  }
+
+  @Before(order = 1, value = "@GCS_OUTPUT_FIELD_TEST")
+  public static void createBucketWithOutputFieldFile() throws IOException, URISyntaxException {
+    gcsSourceBucketName = createGCSBucketWithFile(PluginPropertyUtils.pluginProp("gcsOutputFieldTestFile"));
+  }
+
+  @After(order = 1, value = "@GCS_CSV_TEST or @GCS_TSV_TEST or @GCS_BLOB_TEST " +
+    "or @GCS_DELIMITED_TEST or @GCS_TEXT_TEST or @GCS_OUTPUT_FIELD_TEST")
+  public static void deleteSourceBucketWithFile() throws IOException {
+    deleteGCSBucket(gcsSourceBucketName);
+  }
+
+  @Before(order = 1, value = "@GCS_TARGET_TEST")
+  public static void setTempTargetGCSBucketName() {
+    gcsTargetBucketName = "cdf-e2e-test-" + UUID.randomUUID();
+    BeforeActions.scenario.write("GCS target bucket name - " + gcsTargetBucketName);
+  }
+
+  @After(order = 1, value = "@GCS_TARGET_TEST")
+  public static void deleteTargetBucketWithFile() {
+    deleteGCSBucket(gcsTargetBucketName);
+  }
+
+  @Before(order = 1, value = "@BQ_TARGET_TEST")
+  public static void setTempTargetBQTableName() {
+    gcsBqTargetTable = "E2E_TEST_" + new SimpleDateFormat("yyyy_MM_dd_HH_mm_ss").format(new Date());
+    BeforeActions.scenario.write("BQ Target table name - " + gcsBqTargetTable);
+  }
+
+  @After(order = 1, value = "@BQ_TARGET_TEST")
+  public static void deleteTempTargetBQTable() throws IOException, InterruptedException {
+    try {
+      BigQueryClient.dropBqQuery(gcsBqTargetTable);
+      BeforeActions.scenario.write("BQ Target table - " + gcsBqTargetTable + " deleted successfully");
+    } catch (BigQueryException e) {
+      if (e.getMessage().contains("Not found: Table")) {
+        BeforeActions.scenario.write("BQ Target Table " + gcsBqTargetTable + " does not exist");
+      } else {
+        Assert.fail(e.getMessage());
+      }
+    }
+  }
+
+  @Before(order = 1, value = "@BQ_SOURCE_TEST")
+  public static void createTempSourceBQTable() throws IOException, InterruptedException {
+    gcsBqSourceTable = "E2E_TEST_" + new SimpleDateFormat("yyyy_MM_dd_HH_mm_ss").format(new Date());
+    StringBuilder records = new StringBuilder(StringUtils.EMPTY);
+    for (int index = 2; index <= 25; index++) {
+      records.append(" (").append(index).append(", ").append((int) (Math.random() * (1000) + 1)).append(", '")
+        .append(UUID.randomUUID()).append("'), ");
+    }
+    BigQueryClient.getSoleQueryResult("create table `test_automation." + gcsBqSourceTable + "` as " +
+                                        "SELECT * FROM UNNEST([ " +
+                                        " STRUCT(1 AS Id, 200 as Value, " +
+                                        "'5e8374a9-bbe3-4c29-8878-5d240c58644e' as UID), " +
```

Comment:
what is this hardcoded uuid? Can you add a comment indicating what this query is doing?

Reply:
Replaced hardcoded uuid with random uuid. This is query is creating sample BQ table with random data.

Reply:
@sawantpritam Could you please add a comment with example row describing what type of data will be in the created table?

Reply:
Id - Int, Value - Int, UID - string

Sample row 
Id  | Value | UID 
22 | 968   | 245308db-6088-4db2-a933-f0eea650846a

Reply:
Sorry, I meant in the code so that it looks more readable.

Reply:
Added comment in code.